### PR TITLE
[Main Course] - 租房信息获取

### PR DESCRIPTION
**Pull request by EchoTang04**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Main_Course/One_Day_Build/2021-12-04-renting_data.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+<figure>
+  <img src="https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/img/SH_rent02.PNG" width="500" />
+</figure>
+
+
+## 3 总结
+
+一句话总结，爬取数据的本质就是观察和找规律。那么祝大家找到满意的房子！有时间的话下次写写基于获取的房源信息做的分析，当然，如果鸽了的话当我没说~
+
```
Comment (Tina0330) [inline docs/Main_Course/One_Day_Build/2021-12-04-renting_data.md:156]: LGTM，感谢分享 糖糖。
记得加尾图哦

**Review by Tina0330 — APPROVED** [review]

File: docs/Main_Course/One_Day_Build/2021-12-04-renting_data.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+---
+
+# 获取租房房源数据
+
+!!! info
+    作者：Echo，发布于2021-12-04，阅读时间：约6分钟，微信公众号文章链接：[:fontawesome-solid-link:](https://mp.weixin.qq.com/s/NHgnYaelpcDKYWnAqL6vSw)
+
+## 1 前言
+
+租房/有房要出租的小伙伴们看过来~
```
Comment (jhqian0909) [inline docs/Main_Course/One_Day_Build/2021-12-04-renting_data.md:12]: 可以加一点空行

File: docs/Main_Course/One_Day_Build/2021-12-04-renting_data.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+## 1 前言
+
+租房/有房要出租的小伙伴们看过来~ 
+衣食住行是生活的基本需求。衣和食好解决，不喜欢的衣服可以买新的，不好吃的食物可以换一家吃。可是在住宿上，买房和租房的置换成本都相对较高，因此房源选择尤为慎重。作为目前买不起房的自然人，我们一般是通过中介来实现租房的需求，比如自如，贝壳找房和链家。看来宇宙的尽头是铁岭，租房的尽头是链家……  
+链家占据了租赁市场的主导地位，且提供的信息相对公允。但每当我刷超过十个房源，我就会记不起来每一个的信息，也无法可视化去比较很多个房源。那么让我们开始动手，用万能的Python来让链家变成你家，获取链家网上的你想要的信息吧。（真的不是广告）
+
+
+## 2 获取房源数据
+
+此次侧重XPath的使用和反爬虫小技巧。XPath是用路径表达式在XML文档中选取节点，这里也同样适用于HTML文档的搜索。
```
Comment (jhqian0909) [inline docs/Main_Course/One_Day_Build/2021-12-04-renting_data.md:19]: beautifulsoup似乎也能实现类似的功能

**Review by jhqian0909 — APPROVED** [review]

File: docs/Main_Course/One_Day_Build/2021-12-04-renting_data.md (hunk 1 above)
Comment (EchoTang04) [reply docs/Main_Course/One_Day_Build/2021-12-04-renting_data.md:156]: 好的没问题！

File: docs/Main_Course/One_Day_Build/2021-12-04-renting_data.md (hunk 3 above)
Comment (EchoTang04) [reply docs/Main_Course/One_Day_Build/2021-12-04-renting_data.md:19]: > 

是的，功能一样，但是感觉思路有一点点不一样。beautifulsoup是去直接查找一个pattern，xpath是拆解整个文档结构来找你需要的pattern。我自己觉得后面的更清晰有逻辑哈哈。

File: docs/Main_Course/One_Day_Build/2021-12-04-renting_data.md (hunk 1 above)
Comment (Realvincentyuan) [reply docs/Main_Course/One_Day_Build/2021-12-04-renting_data.md:157]: 尾图有脚本自动加，问题不大

File: docs/Main_Course/One_Day_Build/2021-12-04-renting_data.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    a = get_data(page)
+    #反爬虫策略2：每次爬取随机间隔3-10s
+    time.sleep(random.randint(3,10))
+    count=count+1
+    print ('the '+str(count)+' page is sucessful')
+
+name = ["街区", "租赁方式", "朝向", "每月租金", "行政区","板块","房屋面积","格局","发布时长"]
+page_data = pd.DataFrame( columns= name,data=all_house_list)
+```
+
```
Comment (Realvincentyuan) [inline docs/Main_Course/One_Day_Build/2021-12-04-renting_data.md:148]: 这个结果在手机上能看清楚吗？可以写成markdown table

**Review by Realvincentyuan — APPROVED** [review]
LGTM

File: docs/Main_Course/One_Day_Build/2021-12-04-renting_data.md (hunk 4 above)
Comment (EchoTang04) [reply docs/Main_Course/One_Day_Build/2021-12-04-renting_data.md:148]: 有八百多行，截图就是想举个例子给大家看一下可以pull出来这些数据。等会发预览给大家看一下能不能看清。

**Pull request merged**